protocol/handshake.go

```diff
@@ -0,0 +1,103 @@
+package protocol
+
+import (
+	"github.com/kaspanet/kaspad/addrmgr"
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/netadapter"
+	routerpkg "github.com/kaspanet/kaspad/netadapter/router"
+	peerpkg "github.com/kaspanet/kaspad/protocol/peer"
+	"github.com/kaspanet/kaspad/protocol/receiveversion"
+	"github.com/kaspanet/kaspad/protocol/sendversion"
+	"github.com/kaspanet/kaspad/util/locks"
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+	"sync"
+	"sync/atomic"
+)
+
+func handshake(router *routerpkg.Router, netAdapter *netadapter.NetAdapter, peer *peerpkg.Peer,
+	dag *blockdag.BlockDAG, addressManager *addrmgr.AddrManager) (closed bool, err error) {
+
+	receiveVersionRoute, err := router.AddIncomingRoute([]string{wire.CmdVersion})
```

Comment:
Do we ever remove these routes?
What will happen if peer sends version/verack after handshake is over?